### PR TITLE
MINOR: [Doc][Python] Clarify behavior of pyarrow.compute.fill_value

### PR DESCRIPTION
**Pull request by spenczar**
Discussed in #35624, particularly at https://github.com/apache/arrow/issues/35624#issuecomment-1561345012. 

Doesn't close that issue, but it came up in discussion.

### CODE REVIEW

File: python/pyarrow/compute.py [hunk 1]
```diff
@@ -483,10 +483,16 @@ def take(data, indices, *, boundscheck=True, memory_pool=None):
...
-    """
-    Replace each null element in values with fill_value. The fill_value must be
-    the same type as values or able to be implicitly casted to the array's
-    type.
+    """Replace each null element in values with a corresponding
+    element from fill_value.
+
+    If fill_value is scalar-like, then every null element in values
+    will be replaced with fill_value. If fill_value is array-like,
+    then the i-th element in values will be replaced witth the i-th
```
Comment (pitrou) [inline python/pyarrow/compute.py:491]:
```suggestion
    then the i-th element in values will be replaced with the i-th
```

**Review by pitrou — APPROVED** [review]
Thanks @spenczar ! Just one nit.

**Comment by spenczar** [comment]
Oops! Thanks for directly fixing it :)

**Comment by pitrou** [comment]
CI failures are unrelated.

**Pull request merged**